okhttp/src/main/kotlin/com/connectrpc/okhttp/ConnectOkHttpClient.kt

```diff
@@ -64,11 +66,7 @@ class ConnectOkHttpClient @JvmOverloads constructor(
             newCall.enqueue(
                 object : Callback {
                     override fun onFailure(call: Call, e: IOException) {
-                        val code = if (e.message?.lowercase() == "canceled") {
-                            Code.CANCELED
-                        } else {
-                            Code.UNKNOWN
-                        }
```

Comment:
We handled "cancelled" in a special way here in the unary flow. We handled "timeout" in a special way over in the stream flow. So I consolidated both checks into a new `codeFromIOException` so now both unary and stream calls can benefit from the classification of canceled and deadline exceeded codes.

Reply:
TODO: I can't find anything that actually produces an exception with the word "canceled", so I'm not sure this check really works 🤷. I think most things in the Java ecosystem actually spell it with two Ls, like [`CancellationException`](https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/CancellationException.html)...